libs/datasets/sources/can_scraper_helpers.py

```diff
@@ -83,105 +82,147 @@ def _fips_from_int(param: pd.Series):
-        assert all([getattr(variable, field) for field in required_fields])
-        query_parts = [f"{field} == @variable.{field}" for field in required_fields]
-        for optional_field in ["measurement", "unit"]:
-            if getattr(variable, optional_field):
-                query_parts.append(f"{optional_field} == @variable.{optional_field}")
-
-        return self.all_df.loc[self.all_df.eval(" and ".join(query_parts))].copy()
+    @cached_property
+    def indexed_df(self) -> pd.DataFrame:
+        """The parquet file with many fields moved into a MultiIndex and demographic fields
+        transformed into a single string."""
+        indexed_df = self.all_df.set_index(
+            [
+                Fields.PROVIDER,
+                Fields.VARIABLE_NAME,
+                Fields.MEASUREMENT,
+                Fields.UNIT,
+                Fields.LOCATION_ID,
+            ]
+            + DEMOGRAPHIC_FIELDS
+            + [Fields.DATE]
+        ).sort_index()
+        all_ares = (
+            self.all_df.loc[:, DEMOGRAPHIC_FIELDS]
+            .drop_duplicates()
+            .set_index(DEMOGRAPHIC_FIELDS, drop=False)
+            .apply(make_short_name, axis=1)
+            .rename(PdFields.DEMOGRAPHIC_BUCKET)
+        )
+        # Use `join(other, on=...)` because it preserves the indexed_df index
+        rv = (
+            indexed_df.join(all_ares, on=DEMOGRAPHIC_FIELDS)
+            .set_index(PdFields.DEMOGRAPHIC_BUCKET, append=True)
+            .droplevel(DEMOGRAPHIC_FIELDS)
+        )
+        return rv
 
     def query_multiple_variables(
         self,
         variables: List[ScraperVariable],
         *,
         log_provider_coverage_warnings: bool = False,
         source_type: str,
-    ) -> Tuple[pd.DataFrame, pd.DataFrame]:
-        """Queries multiple variables returning wide df with variable names as columns.
+    ) -> Tuple[pd.Series, pd.DataFrame]:
+        """Queries multiple variables
 
         Args:
             variables: Variables to query
             log_provider_coverage_warnings: Log warnings when upstream data has variables not in
               `variables` and hints when a variable has no data.
+            source_type: String for the `taglib.Source.type` property
 
         Returns:
-            The observations in a DataFrame with variable columns and the source_urls
+            The observations in a Series and the sources in a tag DataFrame
         """
         if log_provider_coverage_warnings:
             self.check_variable_coverage(variables)
 
-        selected_data = []
-        for variable in variables:
-            # Check that `variable` agrees with stuff in the ScraperVariable docstring.
-            if variable.common_field is None:
-                assert variable.measurement == ""
-                assert variable.unit == ""
-                continue
+        # Split `variables` into lists of variables dropped and those returned.
+        variables_drop, variables_return = [], []
```

Comment:
Fwiw I think you could initialize these explicitly 

'''
variables_to_drop = [variable for variable in variables if not variable.common_field]
common_field_variables = [variable for variable in variables if variable.common_field]
'''

Just makes it a bit easier to read the loops